zkir/Utils/AssemblyFormatUtils.cpp

```diff
@@ -28,4 +28,58 @@ void printModulus(AsmPrinter &printer, const APInt &modulus,
   printer << ">";
 }
 
+namespace {
+
+ParseResult doParsePrimeField(OpAsmParser &parser, APInt &parsedInt,
+                              Type &parsedType,
+                              GetModulusCallback getModulusCallback) {
+  if (failed(parser.parseColonType(parsedType))) {
+    return failure();
+  }
+
+  APInt modulus;
+  if (failed(getModulusCallback(modulus))) {
+    return failure();
+  }
+
+  if (parsedInt.getActiveBits() > modulus.getBitWidth()) {
+    return parser.emitError(parser.getCurrentLocation(),
+                            "value is too large for the underlying type");
+  }
+  parsedInt = parsedInt.zextOrTrunc(modulus.getBitWidth());
+  if (failed(validatePrimeField(parser, modulus, parsedInt))) {
+    return failure();
+  }
+  return success();
+}
```

Comment:
how about `parseColonModulusType`

Reply:
Since this function not only parses but also validates input with a prime field, the naming sounds not natural to me. If i make this anyway, then the function will look like as follows, and it seems that it doesn't have a benefit to invoke this function just for parsing..?

```c++
ParseResult parseColonModulusType(OpAsmParser &parser, Type &parsedType,
                                  APInt &modulus,
                                  GetModulusCallback getModulusCallback) {
  if (failed(parser.parseColonType(parsedType))) {
    return failure();
  }

  APInt modulus;
  if (failed(getModulusCallback(modulus))) {
    return failure();
  }

  return success();
}
```